### PR TITLE
List Secret Scan Alerts with index based pagination

### PR DESCRIPTION
**Pull request by AbbanMustafa**

### CODE REVIEW

**Comment by gmlewis** [comment]
Thank you, @AbbanMustafa - were you ever able to test this successfully?

**Comment by AbbanMustafa** [comment]
> Thank you, @AbbanMustafa - were you ever able to test this successfully?

Yes! I verified works for both list option types

**Review by gmlewis — CHANGES_REQUESTED** [review]
Could you please add a unit test demonstrating how/where/why this is being used?

**Comment by gmlewis** [comment]
@AbbanMustafa - it looks like some unit tests broke - could you please fix them?  Also make sure your code is formatted... I haven't taken a look yet... just reporting the breakage.

File: github/secret_scanning.go [hunk 1]
```diff
@@ -61,6 +61,11 @@ type SecretScanningAlertListOptions struct {
 	Resolution string `url:"resolution,omitempty"`
 
 	ListCursorOptions
+
+	// List options can vary on the Enterprise type. On Enterprise Cloud, Secret Scan alerts show to support requesting by page number
+	// along with provided a cursor for an after param https://docs.github.com/en/enterprise-cloud@latest/rest/secret-scanning#list-secret-scanning-alerts-for-an-organization
+	// Whereas on Enterprise Server, it is just by pagination by index https://docs.github.com/en/enterprise-server@3.6/rest/secret-scanning#list-secret-scanning-alerts-for-an-organization
```
Comment (gmlewis) [inline github/secret_scanning.go:67]:
```suggestion
	// List options can vary on the Enterprise type.
	// On Enterprise Cloud, Secret Scan alerts support requesting by page number
	// along with providing a cursor for an "after" param.
	// See: https://docs.github.com/en/enterprise-cloud@latest/rest/secret-scanning#list-secret-scanning-alerts-for-an-organization
	// Whereas on Enterprise Server, pagination is by index.
	// See: https://docs.github.com/en/enterprise-server@3.6/rest/secret-scanning#list-secret-scanning-alerts-for-an-organization
```

**Review by gmlewis — APPROVED** [review]
Thank you, @AbbanMustafa !
LGTM.
Merging.

**Pull request merged**